### PR TITLE
Test editor styles perf

### PR DESCRIPTION
**Pull request by talldan**

### CODE REVIEW

File: packages/block-editor/src/components/editor-styles/index.js [hunk 1]
```diff
@@ -68,14 +68,32 @@ function useDarkThemeBodyClassName( styles, scope ) {
...
+		}, {} );
+
+		// Sort overrides to match the order of blocks they relate to.
+		// This is useful to maintain the correct CSS cascade order for
+		// nested block style variations.
+		const sortedOverrides = [ ...overrides ].sort( ( a, b ) => {
+			const aIndex = clientIdMap[ a[ 1 ].clientId ] ?? -1;
+			const bIndex = clientIdMap[ b[ 1 ].clientId ] ?? -1;
+			return aIndex - bIndex;
+		} );
```
Comment (ellatrix) [inline packages/block-editor/src/components/editor-styles/index.js:92]: Do overrides already have a clientId added to them?

File: packages/block-editor/src/components/editor-styles/index.js [hunk 2]
```diff
@@ -68,14 +68,32 @@ function useDarkThemeBodyClassName( styles, scope ) {
 }
 
 function EditorStyles( { styles, scope } ) {
-	const overrides = useSelect(
-		( select ) => unlock( select( blockEditorStore ) ).getStyleOverrides(),
+	const { overrides, clientIds } = useSelect(
+		( select ) => ( {
+			overrides: unlock( select( blockEditorStore ) ).getStyleOverrides(),
+			clientIds: select( blockEditorStore ).getClientIdsWithDescendants(),
```
Comment (ellatrix) [inline packages/block-editor/src/components/editor-styles/index.js:74]: Does this work for controlled blocks as well?

**Comment by ellatrix** [comment]
For me, this is good, if it solves your style dependency issues

**Comment by talldan** [comment]
@ellatrix Just for some context, this was me pulling some of the code out of [Block Styles: Extend block style variations as mechanism for achieving section styling](https://github.com/WordPress/gutenberg/pull/57908#top) to see what the performance impact of this particular bit of code is.

That PR is causing some performance issues, and it's not clear where right now, but I don't think it's this bit of code.

**Comment by talldan** [comment]
I did notice though that this bit of code now triggers some updates to this component whenever a block is added/moved/removed, not sure there's much impact, but having said that I don't think there's a performance test for block insertion.

File: packages/block-editor/src/components/editor-styles/index.js (hunk 2 above)
Comment (ellatrix) [reply packages/block-editor/src/components/editor-styles/index.js:74]: What if we change `getStyleOverrides` to return the sorted overrides? 🤔

**Comment by ellatrix** [comment]
> Just for some context, this was me pulling some of the code out of https://github.com/WordPress/gutenberg/pull/57908#top to see what the performance impact of this particular bit of code is.

I think it's anyway good to do this separately as a preparatory PR? That PR is huge.

File: packages/block-editor/src/components/editor-styles/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/components/editor-styles/index.js:92]: (I don't see the code in the PR that adds that clientId to the override object)

**Comment by ellatrix** [comment]
I guess these changes were forgotten here?

https://github.com/WordPress/gutenberg/pull/57908/files#diff-cd3303ad473e0bbfc9d92758c62c8f4295b31a0bb8283a3e4445c38fb73c0f4d

**Comment by aaronrobertshaw** [comment]
> I think it's anyway good to do this separately as a preparatory PR? That PR is huge.

Agreed.

FWIW until all the moving parts to achieve the extended block style variations feature were discovered through #57908. We weren't in a position to split this out as the order issues with style overrides were one of the last hurdles found.

Now #57908 has stabilized, there's no reason not to carve out what we can into PRs like this.

> I guess these changes were forgotten here?

Yep. the addition of `clientId` to the style overrides is needed to be able to sort the overrides by the order of blocks and maintain the correct cascade for nested variations.

**Comment by talldan** [comment]
I'll close this as @aaronrobertshaw indicated that [he'll make some new PRs](https://github.com/WordPress/gutenberg/pull/57908#issuecomment-2073693367) with proper descriptions 👍

**Comment by aaronrobertshaw** [comment]
I've put up an alternate PR that includes the addition of `clientId` to style overrides: https://github.com/WordPress/gutenberg/pull/61039.

**Pull request closed without merging**